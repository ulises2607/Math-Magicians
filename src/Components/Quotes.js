import React, { useState, useEffect } from 'react';

function Quotes() {
  const API_KEY = 'PfgoD6eJJrdbX0H0Rpd90w==rNfC856dJmfCqHLG';
  const [quote, setQuote] = useState({ category: '' });
  const [isLoading, setIsLoading] = useState(false);
  const [hasError, setHasError] = useState(null);

  useEffect(() => {
    const fetchData = async () => {
      try {
        setIsLoading(true);
        const res = await fetch('https://api.api-ninjas.com/v1/quotes?category=learning', {
          method: 'GET',
          headers: {
            'X-Api-Key': API_KEY,
            'Content-Type': 'application/json',
          },
        });
        const json = await res.json();
        setQuote(json[0]);
        setIsLoading(false);
      } catch (error) {
        setHasError(error);
        setIsLoading(false);
      }
    };
    fetchData();
  }, []);

  if (isLoading) {
    return <div>Loading...</div>;
  }

  return (
    <div>
      {hasError ? (
        <p>Something went wrong! Please verify the connection.</p>
      ) : (
        <>
          <p className="quote">{quote.quote}</p>
          <span className="quote-author">
            {quote.author}
          </span>
        </>
      )}
    </div>
  );
}

export default Quotes;
