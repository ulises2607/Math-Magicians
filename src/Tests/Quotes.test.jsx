import React from 'react';
import { render, waitFor, screen } from '@testing-library/react';
import Quotes from '../Components/Quotes';

describe('Quotes Component', () => {
  afterEach(() => {
    jest.resetAllMocks();
  });

  test('should display loading state', () => {
    render(<Quotes />);
    expect(screen.getByText('Loading...').textContent).toBe('Loading...');
  });

  it('should display error state', async () => {
    jest.spyOn(global, 'fetch').mockRejectedValue(new Error('API Error'));

    render(<Quotes />);
    await waitFor(() => {
      expect(screen.getByText('Something went wrong! Please verify the connection.').textContent).toBe('Something went wrong! Please verify the connection.');
    });
  });

  test('render', async () => {
    const { quote } = render(<Quotes />);
    await waitFor(() => {
      expect(screen.getByText('Loading...').textContent).toBe('Loading...');
    });
    await waitFor(() => {
      expect(quote).toMatchSnapshot();
    });
  });
});
