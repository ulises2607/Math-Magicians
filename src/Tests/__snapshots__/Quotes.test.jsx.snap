// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quotes Component render 1`] = `undefined`;
